File: frontend/src/services/cityService.js
import { apiClient } from "./apiClient";

export class CityService {

    getCitiesByMovieId(movieId) {
        return apiClient.get("/movie/cities/getCitiesByMovieId/" + movieId);
    }

    getall() {
        return apiClient.get("/movie/cities/getall");
    }

    getTheaterLibrary() {
        return apiClient.get("/movie/cities/theaters/library");
    }

    addCity(cityDto) {
        return apiClient.post("/movie/cities/add", cityDto);
    }

    addTheaterToLibrary(theaterDto) {
        return apiClient.post("/movie/cities/theaters/library", theaterDto);
    }

    addTheater(theaterDto) {
        return apiClient.post("/movie/cities/theaters", theaterDto);
    }

    deleteTheater(movieId, cityId, theaterId) {
        return apiClient.delete(`/movie/cities/${cityId}/theaters/${theaterId}`, {
            params: { movieId }
        });
    }

    deleteCity(movieId, cityId) {
        return apiClient.delete(`/movie/cities/${cityId}`, {
            params: { movieId }
        });
    }
}

File: movieService/src/main/java/com/doffeii/movieService/business/abstracts/CityService.java
package com.doffeii.movieService.business.abstracts;

import com.doffeii.movieService.entity.City;
import com.doffeii.movieService.entity.dto.CityRequestDto;
import com.doffeii.movieService.entity.dto.TheaterRequestDto;

import java.util.List;

public interface CityService {

    List<City> getCitiesByMovieId(int movieId);

    List<City> getTheaterLibrary();

    List<City> getall();

    void add(CityRequestDto cityRequestDto);

    City addTheaterToLibrary(TheaterRequestDto theaterRequestDto, String authorizationHeader);

    City addTheater(TheaterRequestDto theaterRequestDto, String authorizationHeader);

    void deleteTheater(int movieId, int cityId, int theaterId, String authorizationHeader);

    void deleteCityFromMovie(int movieId, int cityId, String authorizationHeader);
}
